lib/puppet/parser/functions/foreman.rb

```diff
@@ -36,6 +36,7 @@
 #
 # Happy Foreman API-ing!
 
+require "pry"
```

Comment:
I see... maybe capturing the failing HTTP request to the foreman API leads to some other hint?